### ISSUE TITLE
Make updater more resilient

### ISSUE THREAD

**Issue by LeanVel** [CybercentreCanada/assemblyline-service-yara#15]
I have just noticed that when there is an issue with a yara rule the updater just crashes and the updater does not proceed with the rest of the signature sources. 

For example I just got the following error:
```json
{"@timestamp": "2021-03-12 13:00:04,506", "event.module": "assemblyline", "event.dataset": "assemblyline.updater.yara", "host.ip": "172.19.0.6", "host.hostname": "ef306476c3ac", "log.level": "ERROR", "log.logger": "assemblyline.updater.yara", "process.pid": "1", "message": "Updater ended with an exception!\nTraceback (most recent call last):\n  File \"/opt/al_service/yara_/yara_updater.py\", line 358, in yara_update\n    signatures = parser.parse_string(\"\\n\".join(temp_lines))\n  File \"/var/lib/assemblyline/.local/lib/python3.7/site-packages/plyara/core.py\", line 291, in parse_string\n    self.parser.parse(input_string, lexer=self.lexer)\n  File \"/var/lib/assemblyline/.local/lib/python3.7/site-packages/ply/yacc.py\", line 333, in parse\n    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)\n  File \"/var/lib/assemblyline/.local/lib/python3.7/site-packages/ply/yacc.py\", line 1120, in parseopt_notrack\n    p.callable(pslice)\n  File \"/var/lib/assemblyline/.local/lib/python3.7/site-packages/plyara/core.py\", line 915, in p_text_strings_kv\n    self._parse_string_kv(p, StringTypes.TEXT)\n  File \"/var/lib/assemblyline/.local/lib/python3.7/site-packages/plyara/core.py\", line 898, in _parse_string_kv\n    raise ParseTypeError(message, p.lineno, p.lexpos)\nplyara.exceptions.ParseTypeError: Duplicate string name key $s5 on line 33\n"}
```

I think that this could be fixed by adding a try / except around the line 358
```python
signatures = parser.parse_string("\n".join(temp_lines))
```
A simple `continue` would at least allow the updater to continue to the next file and not halt the whole updating process.

**Comment by cccs-rs** [comment]
Hello! Thank you for bringing this to our attention and providing a solution. I will begin working on this and should get this patched today! 😁

**Comment by cccs-rs** [comment]
Your fix will be present in versions 4.0.0.stable10 or 4.0.0.dev104. Thanks again for your recommendation! ♥

**Comment by LeanVel** [comment]
Sweet! thanks for the mega quick reaction and fix!
